### PR TITLE
Add E3SM submodule, remove MPAS-Model submodules

### PR DESCRIPTION
**Pull request by xylar**
Update legacy docs with E3SM move

### CODE REVIEW

**Comment by xylar** [comment]
@matthewhoffman, this will have the same issues and discussion as #123.  Let's keep the discussion there for now.

**Comment by xylar** [comment]
@mark-petersen and @matthewhoffman, this is now consistent with #123 and ready for review.  I can't pretend the documentation is perfect but hopefully it's better than before this PR...

File: docs/machine_specific_instructions/linux.rst [hunk 1]
```diff
@@ -84,5 +84,5 @@ Then, when you want to build or run MPAS-Ocean, source a file containing:
     export PNETCDF=${PREFIX}
     export PIO=${PREFIX}
     # change to one of the other cores as needed
-    export CORE=ocean
+    export
```
Comment (matthewhoffman) [inline docs/machine_specific_instructions/linux.rst:87]: This looks like a mistake?  Presumably this whole line should be deleted.

File: docs/ocean/test_cases/isomip_plus_at_lanl.rst [hunk 2]
```diff
@@ -197,7 +197,7 @@ In this file, put:
    export I_MPI_CXX=icpc
    export I_MPI_F77=ifort
    export I_MPI_F90=ifort
-   export CORE=ocean
+   export
```
Comment (matthewhoffman) [inline docs/ocean/test_cases/isomip_plus_at_lanl.rst:200]: same thing here.

File: docs/ocean/test_cases/isomip_plus_at_lanl.rst [hunk 3]
```diff
@@ -237,13 +237,13 @@ Take a look at which branch were on:
 
    git logg
 
-We don't start off on ``MPAS-Dev/MPAS-Model/ocean/develop`` (even though the
+We don't start off on ``E3SM-Project/E3SM/ocean/develop`` (even though the
```
Comment (matthewhoffman) [inline docs/ocean/test_cases/isomip_plus_at_lanl.rst:240]: Is there still an `ocean/develop` branch in E3SM?  Maybe I haven't been paying close enough attention.

File: docs/users_guide/overview.rst [hunk 4]
```diff
@@ -21,17 +21,17 @@ compile MPAS-Ocean:
...
     # load modules (see machine-specific instructions below)
-    make gfortran CORE=ocean
+    make gfortran
 
 For MALI:
 
 .. code-block:: bash
 
-    cd MPAS-Model/landice/develop/
+    cd E3SM-Project/components/mpas-albany-landice
```
Comment (matthewhoffman) [inline docs/users_guide/overview.rst:32]: This one should be `MALI-Dev/components/mpas-albany-landice`

File: docs/users_guide/overview.rst [hunk 5]
```diff
@@ -244,21 +244,27 @@ executable:
...
+     cd components/mpas-ocean
 
-Note that for ocean development, it is best to branch from ``ocean/develop`` and
-for MALI development, start with ``landice/develop``.
+.. note::
+
+    For ocean development, you will clone ``E3SM-Project/E3SM`` and build in
+    ``components/mpas-ocean``; for MALI development, the repo is
+    ``MALI-Dev/E3SM`` and the build directory is
+    ``componets/mpas-albany-landice``.
```
Comment (matthewhoffman) [inline docs/users_guide/overview.rst:270]: typo

**Review by matthewhoffman — APPROVED** [review]
@xylar , I did a quick perusal of the diffs and noted a handful of typos, etc.  Thanks for updating these docs too.  This will be a very helpful resource for the new folks starting this summer.  Hopefully we can get to the v1 release soon and not have to maintain two sets of docs.

File: docs/machine_specific_instructions/linux.rst (hunk 1 above)
Comment (xylar) [reply docs/machine_specific_instructions/linux.rst:87]:
```suggestion
```

File: docs/ocean/test_cases/isomip_plus_at_lanl.rst (hunk 2 above)
Comment (xylar) [reply docs/ocean/test_cases/isomip_plus_at_lanl.rst:200]:
```suggestion
```

File: docs/ocean/test_cases/isomip_plus_at_lanl.rst (hunk 3 above)
Comment (xylar) [reply docs/ocean/test_cases/isomip_plus_at_lanl.rst:240]:
```suggestion
We don't start off on ``E3SM-Project/E3SM/master`` (even though the
```

File: docs/users_guide/overview.rst (hunk 4 above)
Comment (xylar) [reply docs/users_guide/overview.rst:32]:
```suggestion
    cd MALI-Dev/components/mpas-albany-landice
```

File: docs/users_guide/overview.rst (hunk 5 above)
Comment (xylar) [reply docs/users_guide/overview.rst:270]:
```suggestion
    ``components/mpas-albany-landice``.
```

**Comment by xylar** [comment]
@matthewhoffman, sorry about those.  I left a real mess in the ISOMIP+ documentation in particular.  It should be at least mostly better now.  I also hope no one is relying on that documentation too heavily at this point but it should at least make an effort to be useful.

**Review by mark-petersen — APPROVED** [review]
@xylar this looks great! I created the html and read through. I'm sure we will find items to correct as we proceed, but this is a great help, thanks!

**Comment by xylar** [comment]
Thanks @mark-petersen and @matthewhoffman!

**Pull request merged**